### ISSUE TITLE
Plus button background doesn't change when Dark Mode is toggled

### ISSUE THREAD

**Issue by thejayhaykid** [aheze/ProgressGif#12]
If Dark Mode is toggled while in the app, the plus button does not change with it.

## Load into dark mode
![image](https://user-images.githubusercontent.com/9452325/95475545-500f1280-0954-11eb-9c31-99fde1e2c80d.png)

## Switch to light mode
![image](https://user-images.githubusercontent.com/9452325/95475571-5604f380-0954-11eb-8f09-6346bb5409fe.png)

## Close app and reopen in light mode
![image](https://user-images.githubusercontent.com/9452325/95475589-5e5d2e80-0954-11eb-9d6a-e8ff2db6b6e1.png)

**Comment by aheze** [comment]
Nice find! I think this may have to do with the CALayer of the button:
```Swift
class CustomButton: UIButton {
    private var shadowLayer: CAShapeLayer!
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        if shadowLayer == nil {
            shadowLayer = CAShapeLayer()
            shadowLayer.path = UIBezierPath(roundedRect: bounds, cornerRadius: bounds.width / 2).cgPath
            shadowLayer.fillColor = ColorCompatibility.systemBackground.cgColor** /// here!
            
            shadowLayer.shadowColor = ColorCompatibility.secondaryLabel.cgColor
            shadowLayer.shadowPath = shadowLayer.path
            shadowLayer.shadowOffset = CGSize(width: 2.0, height: 2.0)
            shadowLayer.shadowOpacity = 0.4
            shadowLayer.shadowRadius = 3
            
            layer.insertSublayer(shadowLayer, at: 0)
        }
    }
}
```
Maybe the CALayer doesn't react to dark mode? I'll look into it

**Comment by thejayhaykid** [comment]
Want me to take a crack at it?

**Comment by aheze** [comment]
If you want, sure! Check this out: https://stackoverflow.com/a/57177411/14351818

**Comment by thejayhaykid** [comment]
I'll take it then

**Comment by aheze** [comment]
I've found a fix! Instead of 
```
override func tintColorDidChange() {
```
Just do:
```
override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
```

**Comment by aheze** [comment]
Fixed in 94deacd834a66964e3b041896ef306739b4e73b8. Again, thanks for finding this bug @thejayhaykid!

**Comment by thejayhaykid** [comment]
No problem! I was glad to help
